modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/AttributionDocumentGeneratorImpl.java

```diff
@@ -72,15 +72,29 @@ public AttributionDocumentGeneratorImpl(String documentName, File workingDir, St
      * @return (non - null) the file handle of the generated attribution document.
      */
     public File generate(List<ArtifactAndLicense> artifacts) {
-        Templates templates = TemplateLoaderUtil.load(templateKey);
-
-        File title = writeTitle(templates);
-        File copyright = writeCopyright(templates);
-        File backPage = writeBackPage(templates.getBackPage());
-        File artifactPages = writeArtifacts(templates, artifacts, templates.getContent());
-        File intermediateDoc = mergePages(title, copyright, artifactPages, backPage);
-
-        return postProcess(templates, intermediateDoc, documentName);
+        Templates templates = null;
+        try {
+            templates = TemplateLoaderUtil.load(templateKey);
+
+            File title = writeTitle(templates);
+            File copyright = writeCopyright(templates);
+            File backPage = writeBackPage(templates.getBackPage());
+            File artifactPages = writeArtifacts(templates, artifacts, templates.getContent());
+            File intermediateDoc = mergePages(title, copyright, artifactPages, backPage);
+
+            return postProcess(templates, intermediateDoc, documentName);
+        } finally {
+            if (templates != null) {
+                try {
+                    templates.getBackPage().close();
```

Comment:
I think, we should close each page in an own try catch, and only log, if there is an IOException here. It is not critical to the execution if the closing fails, but it should be remarked.